packages/@ember/-internals/views/lib/views/core_view.ts

```diff
@@ -20,21 +19,17 @@ import states from './states';
   @uses Ember.ActionHandler
   @private
 */
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-interface CoreViewClass extends CoreObjectClass<CoreView> {}
-interface CoreView extends FrameworkObject, Evented, ActionHandler, View {}
-const CoreView = (FrameworkObject.extend(Evented, ActionHandler, {
-  isView: true,
 
-  _states: states,
+interface CoreView extends FrameworkObject, Evented, ActionHandler, View {}
+class CoreView extends FrameworkObject.extend(Evented, ActionHandler) {
+  isView = true;
 
-  init() {
-    this._super(...arguments);
-    this._state = 'preRender';
-    this._currentState = this._states.preRender;
-  },
+  _states = states;
```

Comment:
One important note: this has meaningfully different semantics than what it replaced: it's now one reference per instance to the states object, rather than a single reference on the prototype. While in this case it should be a *small* impact, it's not zero, especially for `CoreView`. We should see if we can come up with a better way of doing this to avoid that. :thinking:

Reply:
Good catch. I’ll review